devices/sinope.js

```diff
@@ -972,8 +993,11 @@ module.exports = [
         description: 'Zigbee smart light switch',
         fromZigbee: [fz.on_off, fz.electrical_measurement, fzLocal.sinope],
         toZigbee: [tz.on_off, tzLocal.timer_seconds, tzLocal.led_intensity_on, tzLocal.led_intensity_off,
-            tzLocal.led_color_on, tzLocal.led_color_off],
+            tzLocal.led_color_on, tzLocal.led_color_off, tzLocal.keypad_lock],
         exposes: [e.switch(),
+            exposes.enum('action_report', ea.STATE, ['up_single', 'up_double', 'up_hold',
```

Comment:
I missed this on my initial review but please rename this to `action`  and use `e.action()` (e.g. https://github.com/Koenkk/zigbee-herdsman-converters/blob/2861a72762fab96952a9077bb1551770a53b3200/devices/xiaomi.js#L871)

Reply:
Here you go.  In my first attempt, I used `e.action`, but then I hesitate since `action` is also used when the option  `state_action` is active.  I asked the question in the [discussions](https://github.com/Koenkk/zigbee2mqtt/discussions/17287), but no one replied.  By the way, thanks for your great dedication.